node/src/components/contract_runtime.rs

```diff
@@ -158,18 +202,71 @@ where
     REv: From<ContractRuntimeRequest>
         + From<ContractRuntimeAnnouncement>
         + From<ControlAnnouncement>
+        + From<NetworkRequest<NodeId, Message>>
         + Send,
 {
-    type Event = ContractRuntimeRequest;
+    type Event = Event;
     type ConstructionError = ConfigError;
 
     fn handle_event(
         &mut self,
         effect_builder: EffectBuilder<REv>,
-        _rng: &mut NodeRng,
-        event: Self::Event,
+        rng: &mut NodeRng,
+        event: Event,
     ) -> Effects<Self::Event> {
         match event {
+            Event::ContractRuntimeRequest(request) => {
+                self.handle_contract_runtime_request(effect_builder, rng, *request)
+            }
+            Event::TrieRequestIncoming(request) => {
+                self.handle_trie_request(effect_builder, request)
+            }
+        }
+    }
+}
+
+impl ContractRuntime {
+    fn handle_trie_request<REv>(
+        &self,
+        effect_builder: EffectBuilder<REv>,
+        TrieRequestIncoming {
+            sender,
+            message: TrieRequest(ref serialized_id),
+        }: TrieRequestIncoming,
+    ) -> Effects<Event>
+    where
+        REv: From<NetworkRequest<NodeId, Message>> + Send,
+    {
+        let fetched_or_not_found = match self.get_trie(serialized_id) {
+            Ok(fetched_or_not_found) => fetched_or_not_found,
+            Err(error) => {
+                debug!("failed to get trie: {}", error);
+                return Effects::new();
+            }
+        };
+
+        match Message::new_get_response(&fetched_or_not_found) {
+            Ok(message) => effect_builder.send_message(sender, message).ignore(),
+            Err(error) => {
+                error!("failed to create get-response: {}", error);
+                Effects::new()
+            }
+        }
+    }
+
+    fn handle_contract_runtime_request<REv>(
```

Comment:
Same as above, just a single line doc comment for completeness sake?

Reply:
Added in [9d1fd9c0687d181ac7d71c02941c19630e4e6b04](https://github.com/casper-network/casper-node/pull/2614/commits/9d1fd9c0687d181ac7d71c02941c19630e4e6b04)